### ISSUE TITLE
Crash on startup before skywanderers logo

### ISSUE THREAD

**Issue by NoxyNixie** [tsunamayo/Skywanderers#317]
My moms PC crashes before the logo appears but after the unity logo.

Her PC isn't the best but the GPU is okayish and should supports DirectX 11:

- Windows 7
- AMD Athlon 7850 Dual Core @ 2.8GHz
- 4 GB Ram
- Nvidia GT 720 2GB

[2017-10-31_101046.zip](https://github.com/tsunamayo/Skywanderers/files/1430109/2017-10-31_101046.zip)

The graphics drivers do seem to crash even after updating them to the latest version available.

This happens at a clean install even after clearing everything in `\AppData\LocalLow\TsunamayoCorp\*`.

I even tried extracting Characters.zip which you've provided in other bug reports but this had no effect at all (note that the game never made any files related to characters in the appdata folder).

**Comment by Napther** [comment]
try booting the game at Minimum resolution, and keep clicking the mouse during the "made with Unity" screen untill you crash or get into the game then change the resolution back. Identical to my issue a week+ ago of #118

**Comment by CyaNox** [comment]
I've had some time to test a bit further and tried various things including the suggestions from @Napther but there was no change at all.

I've even forcefully disabled or enabled various features through the nvidia control panel and nothing had an effect. I'm fairly confident there is not much I can do at this time to make it actually load up to the initial menu.

**Comment by CaptainCJH** [comment]
Try opening the game in windowed or full whichever is opposite of your original choice, it will crash. Then attempt your original setting and see if it now works. I think doing those steps somehow made it work for me with a low end GFX card a GT 730 with 1 gb VRAM and I want to verify. Let me know here if it works.

**Comment by CyaNox** [comment]
@CaptainCJH Nope it always crashes. Fullscreen, windowed, in any order.

**Comment by CaptainCJH** [comment]
Maybe keep trying. I think after about 6 or more crashes it somehow worked for me. Same happened with previous version. Im not sure then why it worked for me after multiple crashes but try that. There has to be a reason it now works for me.

**Comment by CyaNox** [comment]
With all the testing I did it has crashed probably more than 30 times ... don't think stubbornness is going to be the solution on this one.

**Comment by CaptainCJH** [comment]
Hmm, well now that I think about it the game worked for me when I choose a screen size that was one size smaller than my desktop in windowed mode. Maybe find out your resolution by right clicking anywhere on desktop screen then choose a screen size one smaller than that. I believe I am using 1600x900 windowed with a desktop slighly bigger in width and height. I have to get home from work to be entirely certain.

**Comment by CyaNox** [comment]
I appreciate your efforts in trying to help but I'm very familiar with computers (and programming) having worked with them for over 25 years now. I've tried various resolutions going from smallest possible up to maximum possible both windowed and fullscreen.

**Comment by CaptainCJH** [comment]
There has to be a reason why it now works on mine though. I had same issue. I will check more later when I get home.

**Comment by CaptainCJH** [comment]
Still have not found any connection to the crash. All I know at this point is that my PC is running at 1920x1080 desktop. I can run game with 1600x900 windowed or full screen. When I changed to a lower resolution the game crashed. When I set it back to 1600x900 it worked ok. I also was able to set VSYNC to ON in the main menu OPTIONS then GRAPHICS when I did get it to run. Not sure where this setting is saved but I read @Napther comments and that may be something he did too.

**Comment by CaptainCJH** [comment]
You must have gotten the game to work since I see.you posted the speedometer bug. I tried to reproduce how I got mine to work and now cant get it to work without crashing lol. There seems to be an issue with changing resolutions. Although I briefly got mine to work at 640x480 windowed then choose 1600x1900 windowed when I exited to desktop then changed to 1600x900 it started crashing again. Hope this helps you @tsunamayo

**Comment by CyaNox** [comment]
@CaptainCJH My mom's computer is the one that is affected by the crash I've described not my own.

**Comment by tsunamayo** [comment]
Okay guys I will provide a custom build with a few option off to investigate these crash, as I am not reproducing them. I keep you posted. Thanks for your patience.

**Comment by tsunamayo** [comment]
Hi, I made some change to make startup easier on the GFX card. Could you please activate the debug version inside steam (right click on the game, then properties, then betas, then choose debug), and tell me if the game boot correctly? Thanks!

**Comment by CyaNox** [comment]
Initial testing (with "dirty" leftover settings file was successful though the memory usage did shoot up to maximum while it was generating the world.

[output_log.txt](https://github.com/tsunamayo/Skywanderers/files/1474322/output_log.txt)

In the log the following is repeated a lot:
```
DllNotFoundException: AsyncTextureReader
  at (wrapper managed-to-native) AsyncTextureReader:RequestBufferData (intptr)
  at AsyncTextureReader.RequestBufferData (UnityEngine.ComputeBuffer buffer) [0x00000] in <filename unknown>:0 
  at TorusLeaf.GenerateHeights () [0x00000] in <filename unknown>:0 
  at TorusLeaf.Start () [0x00000] in <filename unknown>:0 
  at TorusNode.SetTorus (.TorusHabitat torus, Vector3 position, Int32 level, Int32 c_i, Int32 d_i) [0x00000] in <filename unknown>:0 
  at TorusLeaf.SetTorus (.TorusHabitat torus, Vector3 position, Int32 level, Int32 c_i, Int32 d_i) [0x00000] in <filename unknown>:0 
  at TorusHabitat.CreateNode (Int32 level, Int32 c_i, Int32 d_i) [0x00000] in <filename unknown>:0 
  at TorusQuadNode.CreateChildren (Int32 i, Int32 ci_step, Int32 cd_step) [0x00000] in <filename unknown>:0 
  at TorusQuadNode.Split () [0x00000] in <filename unknown>:0 
  at TorusNode.CheckSplit (Vector3 camPos) [0x00000] in <filename unknown>:0 
  at TorusQuadNode.CheckSplitChildrens (Vector3 camPos) [0x00000] in <filename unknown>:0 
  at TorusNode.CheckSplit (Vector3 camPos) [0x00000] in <filename unknown>:0 
  at TorusQuadNode.CheckSplitChildrens (Vector3 camPos) [0x00000] in <filename unknown>:0 
  at TorusNode.CheckSplit (Vector3 camPos) [0x00000] in <filename unknown>:0 
  at TorusQuadNode.CheckSplitChildrens (Vector3 camPos) [0x00000] in <filename unknown>:0 
  at TorusNode.CheckSplit (Vector3 camPos) [0x00000] in <filename unknown>:0 
  at TorusQuadNode.CheckSplitChildrens (Vector3 camPos) [0x00000] in <filename unknown>:0 
  at TorusNode.CheckSplit (Vector3 camPos) [0x00000] in <filename unknown>:0 
  at TorusHabitat.CheckNode (Int32 c, Int32 d, Quaternion r, Vector3 camPos) [0x00000] in <filename unknown>:0 
  at TorusHabitat.UpdateDepthLine (Int32 c, Quaternion r, Vector3 camPos) [0x00000] in <filename unknown>:0 
  at TorusHabitat.LateUpdate () [0x00000] in <filename unknown>:0 
```

This did not affect the loading of the world however beyond looking around in the game I did not move around to inspect if things were fine. Second test will follow.

**Comment by tsunamayo** [comment]
Hi, thanks for testing! It is only supposed to have improved the loading issue, not the game yet.
Your error is coming from a DLL I am using to fetch GPU data without stalling the CPU. I will try to investigate what is wrong with that. What is your windows 7 version (SP1?), and directx ? (11, 11.1 ?) Thanks

**Comment by CyaNox** [comment]
Second test results:

I wanted to do a clean test but I found I had to also clear the registry: `HKEY_CURRENT_USER\Software\TsunamayoCorp\Skywanderers` as well as `%AppData%\..\LocalLow\TsunamayoCorp\Skywanderers`. After doing so the game started up as previously. This time I tried walking around and manage to look outside of the starter building and saw the ringworld was not visible. then I turned around and I saw a portion of the ringworld but at the same time the game froze. I had to kill it to continue testing.

The `Nvidia GT 720` is a DirectX 12 compatible card so should support DirectX 11 fully:
![gpu](https://user-images.githubusercontent.com/211381/32833619-b2985890-c9ff-11e7-9eab-e22199376dd1.gif)

The OS: Windows 7 64-bit SP1

**Comment by CyaNox** [comment]
I seem to not get past the huge memory spike issue as portrayed in:
![sw_memory_peak](https://user-images.githubusercontent.com/211381/32834452-f252d2a0-ca02-11e7-90ca-0b7bc1288516.png)

In this example I killed it at the top of the peak because I knew it wouldn't go well. On my own system Skywanderers tends to not go much above 1GB so I know something is off.

**Comment by tsunamayo** [comment]
Hi, okay it seems that now the startup is fixed you have an issue with the AsyncTextureReader dll I am using.
For some reason it seems that the system is not able to load it. Some other user had this problem, and one got it fixed but was not able to tell me what changed on his system.

Could you please run the following software on that dll and tell me what are you seing?? 
http://www.dependencywalker.com/
Dll is located inside \Skywanderers_Data\Plugins\AsyncTextureReader.dll
Here is what I get on my pc: 
[loading.txt](https://github.com/tsunamayo/Skywanderers/files/1477728/loading.txt)
[not loading.txt](https://github.com/tsunamayo/Skywanderers/files/1477729/not.loading.txt)

(if curious you can view more info about the dll over here ;-) , I need it to deal with fetch collision data from the gpu without stalling the cpu) 
https://github.com/SlightlyMad/AsyncTextureReader

Thanks again for your help!

**Comment by CyaNox** [comment]
It is missing the following dll's which seem to indicate missing runtime.

```
API-MS-WIN-CRT-HEAP-L1-1-0.DLL
API-MS-WIN-CRT-RUNTIME-L1-1-0.DLL
MSVCP140.DLL
VCRUNTIME140.DLL
```

After googling a bit I found that it seems to point to the VC runtime of 2015 so I downloaded that from: https://www.microsoft.com/en-gb/download/details.aspx?id=48145

After installation of this the dependency walker shows much promise.

I'm about to test if it fixes the loading issues (and have to close the browser due to memory consumption) but I will report back shortly.
